exec/plugin-commands-script-runners/src/shouldRunCheck.ts

```diff
@@ -19,4 +12,4 @@ const SCRIPTS_TO_SKIP = [
   'postuninstall',
 ]
 
-export const shouldRunCheck = (env: Env, scriptName: string): boolean => !env[SKIP_ENV_KEY] && !SCRIPTS_TO_SKIP.includes(scriptName)
+export const shouldRunCheck = (env: Env): boolean => !EVENTS_TO_SKIP.includes(env.npm_lifecycle_event)
```

Comment:
Isn't it fine to do this
```suggestion
export const shouldRunCheck = (env: Env): boolean => env.npm_lifecycle_event != null
```

We won't ever verify deps in nested scripts anyway (because we set verify-deps-before-run to false) and npm_lifecycle_event isn't defined when you run the first `pnpm run`